File: commands/lookup.js
const { Command, Middleware } = require('yamdbf');
const { resolve, expect } = Middleware;
const utils = require('../utils');

module.exports = class extends Command {
	constructor() {
		super({
			name: 'lookup',
			desc: 'Returns the Discord Username associated with this Forum Profile',
			usage: '<prefix>lookup <forum_username>',
		});

		this.use(resolve('forum_username?: String'));
	}

	action(message, args) {
		// Lookup discord user associated with provided forum username, or self if no arguments
        utils.post('lookup.php', { forum_username: args[0] }, body => {
            try {
                let username = message.client.users.find("id", body).username;
                message.reply('This account is linked to: ' + username);
            } catch(err) {
                message.reply(body);
            }
        });
	}
};

File: commands/lookup.py
from commands.base_command  import BaseCommand
import requests
import settings
from utils                  import get_base_url, get_emoji

class Lookup(BaseCommand):
    def __init__(self):
        description = "Returns the Discord Username associated with this Forum Profile"
        params = ['forum_username']
        super().__init__(description, params)

    async def handle(self, params, message, client):
        url = get_base_url() + 'lookup.php'
        data = {
            'forum_username': ' '.join(params),
        }
        r = requests.post(url, data)
        
        try:
            print('r.text: ' + r.text)
            user = await client.get_user_info(r.text)
            print(dir(user))
            msg = 'This account is linked to: ' + user.name
        except Exception as e:
            print(e)
            msg = 'Error: ' + r.text
        
        await client.send_message(message.channel, msg)

